src/components/Dialog/index.tsx

```diff
@@ -105,19 +104,34 @@ export function Outer({
     [setOpenIndex, setDialogIsOpen, control.id],
   )
 
+  // This is the function that we call when we want to dismiss the dialog.
   const close = React.useCallback<DialogControlProps['close']>(cb => {
-    if (cb && typeof cb === 'function') {
-      if (closeCallback.current) {
-        logger.error(
-          `Dialog close was passed multiple callbacks, you shouldn't do that`,
-        )
-      }
+    if (typeof cb === 'function') {
       closeCallback.current = cb
     }
-    // initiates a close animation, the actual "close" happens in `onCloseInner`
     sheet.current?.close()
   }, [])
```

Comment:
Do we actually care if the callback changes? Or can it? From what I can tell if we call `close` then the dialog's lifetime is going to end and there won't be another chance for this to get called. Unless we are worried about `control.close()` getting called multiple times for some reason?

Reply:
In principle we should take multiple calls into account since there's no mechanism to prevent those, and if they happen in different parts of the codebase (which at first might be independent and then due to another change might start firing together), each of them may have an expectation that the callback will fire.

Reply:
Okay. Perhaps I am misunderstanding what is going on outside of here, but I would imagine that we wouldn't actually hit this case. Let me look some more. I'd rather that just not be a possibility at all.